src/mono/mono/metadata/assembly.c

```diff
@@ -1783,15 +1793,15 @@ mono_assembly_load_reference (MonoImage *image, int index)
 	 * image->references is shared between threads, so we need to access
 	 * it inside a critical section.
 	 */
-	mono_assemblies_lock ();
+	mono_image_references_lock (image);
```

Comment:
There is already a mono_image_lock () low level lock.

Reply:
@vargaz what are the reasons to prefer using the image lock instead of having a separate lock just for this field?

Reply:
@vargaz ping